tasks/prepare-binaries.js

```diff
@@ -71,28 +71,38 @@ function copyAssets (platforms) {
 
 async function archive (platforms) {
   console.log('Zipping...')
-  await Promise.all(Object.keys(platforms).map(async (platform) => {
+
+  const promises = Object.keys(platforms).map(async (platform) => {
+    const rootname = `${appName}-${platform}`
     const archive = archiver('zip', {
       zlib: { level: 9 } // Maximize compression
     })
 
-    // must not be in same dir where we are zipping
-    const zipOut = fs.createWriteStream(path.join(buildDirPath, `${appName}-${platform}.zip`))
-    zipOut.on('close', function () {
-      console.log(`Wrote ${Math.round(archive.pointer() / 1e4) / 1e2} Mb total to ${platform}`)
-    })
-    archive.on('error', function (err) {
-      throw err
-    })
+    const zipOut = fs.createWriteStream(path.join(buildDirPath, `${rootname}.zip`))
+
     // pipe archive to file stream
     archive.pipe(zipOut)
+
+    // must not be in same dir where we are zipping
+    const p0 = new Promise((resolve, reject) => {
+      zipOut.on('close', function () {
+        console.log(`Wrote ${Math.round(archive.pointer() / 1e4) / 1e2} Mb total to ${platform}`)
+        resolve()
+      })
+      zipOut.on('error', reject)
+      archive.on('error', reject)
+    })
+
     // recursively add directory to _root_ of zip
-    archive.directory(path.join(buildDirPath, platform), false)
-    await archive.finalize()
-  }))
+    archive.directory(path.join(buildDirPath, platform), rootname, {})
```

Comment:
We need to update the smoke test to take into account the root directory: https://github.com/Mogztter/asciidoctor-web-pdf/blob/36b4d59654742d357ff0a096355cc552e4ff2480/.github/workflows/build.yml#L74-L78

Reply:
Good point. This change is a mistake by me. It should be in a second PR strictly about #456 while this PR is about "why the hell is `npm run build` not exiting automatically.